fmriprep/utils/transforms.py

```diff
@@ -36,57 +37,76 @@ def load_transforms(xfm_paths: list[Path], inverse: list[bool]) -> nt.base.Trans
     return chain
 
 
-def load_ants_h5(filename: Path) -> nt.TransformChain:
-    """Load ANTs H5 files as a nitransforms TransformChain"""
-    affine, warp, warp_affine = parse_combined_hdf5(filename)
-    warp_transform = nt.DenseFieldTransform(nb.Nifti1Image(warp, warp_affine))
-    return nt.TransformChain([warp_transform, nt.Affine(affine)])
+FIXED_PARAMS = np.array([
+    193.0, 229.0, 193.0,  # Size
+    96.0, 132.0, -78.0,   # Origin
+    1.0, 1.0, 1.0,        # Spacing
+    -1.0, 0.0, 0.0,       # Directions
+    0.0, -1.0, 0.0,
+    0.0, 0.0, 1.0,
+])  # fmt:skip
 
 
-def parse_combined_hdf5(h5_fn, to_ras=True):
+def load_ants_h5(filename: Path) -> nt.TransformChain:
+    """Load ANTs H5 files as a nitransforms TransformChain"""
     # Borrowed from https://github.com/feilong/process
     # process.resample.parse_combined_hdf5()
-    h = h5py.File(h5_fn)
+    #
+    # Changes:
+    #   * Tolerate a missing displacement field
+    #   * Return the original affine without a round-trip
+    #   * Always return a nitransforms TransformChain
+    #
+    # This should be upstreamed into nitransforms
+    h = h5py.File(filename)
     xform = ITKCompositeH5.from_h5obj(h)
-    affine = xform[0].to_ras()
+
+    # nt.Affine
+    transforms = [nt.Affine(xform[0].to_ras())]
+
+    if '2' not in h['TransformGroup']:
+        return transforms[0]
+
+    transform2 = h['TransformGroup']['2']
+
     # Confirm these transformations are applicable
-    assert (
-        h['TransformGroup']['2']['TransformType'][:][0] == b'DisplacementFieldTransform_float_3_3'
-    )
-    assert np.array_equal(
-        h['TransformGroup']['2']['TransformFixedParameters'][:],
-        np.array(
-            [
-                193.0,
-                229.0,
-                193.0,
-                96.0,
-                132.0,
-                -78.0,
-                1.0,
-                1.0,
-                1.0,
-                -1.0,
-                0.0,
-                0.0,
-                0.0,
-                -1.0,
-                0.0,
-                0.0,
-                0.0,
-                1.0,
-            ]
-        ),
-    )
+    if transform2['TransformType'][:][0] != b'DisplacementFieldTransform_float_3_3':
+        msg = 'Unknown transform type [2]\n'
+        for i in h['TransformGroup'].keys():
+            msg += f'[{i}]: {h["TransformGroup"][i]["TransformType"][:][0]}\n'
+        raise ValueError(msg)
+
+    fixed_params = transform2['TransformFixedParameters'][:]
+    if not np.array_equal(fixed_params, FIXED_PARAMS):
+        msg = 'Unexpected fixed parameters\n'
+        msg += f'Expected: {FIXED_PARAMS}\n'
+        msg += f'Found: {fixed_params}'
+        if not np.array_equal(fixed_params[6:], FIXED_PARAMS[6:]):
+            raise ValueError(msg)
+        warnings.warn(msg)
```

Comment:
is the purpose of warning than erroring to allow use of customized xform parameters?

Reply:
The purpose is for debugging. I *think* we know how to deal with unusual parameters, but it would be nice to get the hint that this could be the source of problems, if something goes wrong.